internal/pkg/config/env_defaults.go

```diff
@@ -156,19 +159,27 @@ func init() {
 	}
 }
 
-func loadLimits() *envLimits {
-	ramSize := int(memory.TotalMemory() / 1024 / 1024)
-	return loadLimitsForRam(ramSize)
+func initLimits() *envLimits {
+	return loadLimits(0)
 }
 
-func loadLimitsForRam(currentRAM int) *envLimits {
+func loadLimits(agentLimit int) *envLimits {
+	return loadLimitsForAgents(agentLimit)
+}
+
+func loadLimitsForAgents(agentLimit int) *envLimits {
 	for _, l := range defaults {
-		// get max possible config for current env
-		if l.RAM.Min < currentRAM && currentRAM <= l.RAM.Max {
+		// get nearest limits for configured agent numbers
+		if l.Agents.Min < agentLimit && agentLimit <= l.Agents.Max {
+			log.Info().Msgf("Using system limits for %d to %d agents for a configured value of %d agents", l.Agents.Min, l.Agents.Max, agentLimit)
+			ramSize := int(memory.TotalMemory() / 1024 / 1024)
+			if ramSize < l.RecommendedRAM {
+				log.Warn().Msgf("Detected %d MB of system RAM, which is lower than the recommended amount (%d MB) for the configured agent limit", ramSize, l.RecommendedRAM)
```

Comment:
i attempted to pass the logger down from the call point, but the logger requires the config to be parsed and loaded before it knows how to configure itself. if we are concerned about consistency of logs (i.e. these won't be using the ecs schema), then it will be worth loading the logging config first before the rest of the configuration file. thoughts?

similarly, since this code first runs before the config is first built, these log lines will always show up regardless of configuration settings. will this be potentially confusing?

Reply:
This might be indeed confusing for the users, Looking at the logging and how its used quickly I think some work might be needed to clean up the logging initialization.  I would prefer to have log and create an issue for clean up the log in a more general way, WDYT @ruflin?

Reply:
There definitively seems to be a smell here.

@ph As you know, elastic-agent and fleet-server have a bit a different history. If we can align logging would be great.